src/tracer_options.cpp

```diff
@@ -3,25 +3,108 @@
 #include <iterator>
 #include <sstream>
 
+#include <datadog/tags.h>
+#include <opentracing/ext/tags.h>
+
 namespace ot = opentracing;
 
 namespace datadog {
 namespace opentracing {
 
+// Extracts key-value pairs from a string.
+// Duplicates are overwritten.
+// Intended use is for settings tags from DD_TAGS options.
+std::map<std::string, std::string> keyvalues(std::string text, char itemsep, char tokensep,
+                                             char escape) {
+  // early-return if empty
+  if (text.empty()) {
+    return {};
+  }
+
+  bool esc = false;
+  std::map<std::string, std::string> kvp;
+  std::string key;
+  std::string val;
+  bool keyfound = false;
+  auto assignchar = [&](char c) {
+    if (keyfound) {
+      val += c;
+    } else {
+      key += c;
+    }
+    esc = false;
+  };
+  auto addkv = [&](std::string key, std::string val) { kvp[key] = val; };
+  for (auto ch : text) {
+    if (esc) {
+      assignchar(ch);
+      continue;
+    }
+    if (ch == escape) {
+      esc = true;
+      continue;
+    }
+    if (ch == tokensep) {
+      addkv(key, val);
+      key = "";
+      val = "";
+      keyfound = false;
+      continue;
+    }
+    if (ch == itemsep) {
+      keyfound = true;
+      continue;
+    }
+    assignchar(ch);
+  }
+  if (!key.empty()) {
+    addkv(key, val);
+  }
+
+  return kvp;
+}
+
 ot::expected<TracerOptions, const char *> applyTracerOptionsFromEnvironment(
     const TracerOptions &input) {
   TracerOptions opts = input;
 
-  auto agent_host = std::getenv("DD_AGENT_HOST");
-  if (agent_host != nullptr && std::strlen(agent_host) > 0) {
-    opts.agent_host = agent_host;
-  }
-
   auto environment = std::getenv("DD_ENV");
   if (environment != nullptr && std::strlen(environment) > 0) {
     opts.environment = environment;
   }
 
+  auto service = std::getenv("DD_SERVICE");
+  if (service != nullptr && std::strlen(service) > 0) {
+    opts.service = service;
+  }
+
+  auto version = std::getenv("DD_VERSION");
+  if (version != nullptr && std::strlen(version) > 0) {
+    opts.version = version;
+  }
+
+  auto tags = std::getenv("DD_TAGS");
+  if (tags != nullptr && std::strlen(tags) > 0) {
+    opts.tags = keyvalues(tags, ':', ',', '\\');
+    // Special case for env and sampling priority
```

Comment:
(and version)

Reply:
also, interesting, we just ignore any `env`/`version` set via `DD_TAGS`?

Reply:
not sure we need to, should be okay to allow this (other languages allow it)

Reply:
Yeah, I might change the behavior/precedence in the future.
The current behavior is to ignore `environment` in `DD_TAGS` if `DD_ENV` is also set. (and same for `version`).

There's currently 4 sources for some settings:
- environment variable
- JSON config file
- TracerOptions struct (defaults but user's code can change)
- hard-coded defaults

Picking the right result is tricky when `DD_TAGS` gets involved. I'd prefer if there were certain tags that can't be set via DD_TAGS to make things simpler. What's there at the moment is a half-way measure instead of fully ignoring certain keys.